functions/sync-review-comments-db/models.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import enum
+from datetime import datetime
+from typing import List, Optional
+
+from sqlalchemy import ForeignKey, ScalarResult, UniqueConstraint, func, select
+from sqlalchemy.orm import (
+    DeclarativeBase,
+    Mapped,
+    Session,
+    mapped_column,
+    relationship,
+    selectinload,
+)
+
+
+class EvaluationAction(enum.Enum):
+    APPROVE = 1
+    REJECT = 2
+    IGNORE = 3
+
+
+class DiffStatus(enum.Enum):
+    PENDING = 1
+    GENERATED = 2
+    IGNORED = 3
+    SUBMITTED = 4
+    REPLACED = 5
+
+
+class IgnoreReasons(enum.Enum):
+    SIZE = 1
+
+
+class SuggestionIgnoreReason(enum.Enum):
+    NOT_SURE = 1
+    TRIVIAL = 2
+    DEVELOPMENT_PHASE = 3
+    OTHER = 4
+    REVIEW_TIP = 5
+    INCORRECT = 6
+    VALID_REDUNDANT = 7
+
+
+class ReviewRequestMode(enum.Enum):
+    NORMAL = 1
+    EXPERIMENTAL = 2
+
+
+class Base(DeclarativeBase):
+    pass
+
+
+class ReviewRequest(Base):
+    __tablename__ = "review_requests"
+    __table_args__ = (UniqueConstraint("diff_id", "mode", "sequence"),)
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    diff_id: Mapped[int] = mapped_column(index=True)
+    revision_id: Mapped[int] = mapped_column(index=True)
+    status: Mapped[DiffStatus]
+    suggestions: Mapped[List["Suggestion"]] = relationship(
+        back_populates="review_request"
+    )
+    bugbug_version: Mapped[str]
+    tool_variant: Mapped[Optional[str]]
+    ignore_reason: Mapped[Optional[IgnoreReasons]]
+    mode: Mapped[ReviewRequestMode]
+    sequence: Mapped[int]
+
+    # pylint:disable=not-callable
+    created_at: Mapped[datetime] = mapped_column(server_default=func.now())
+    updated_at: Mapped[datetime] = mapped_column(
+        server_default=func.now(), server_onupdate=func.now()
+    )
+
+    @property
+    def is_recently_created(self):
+        return (datetime.utcnow() - self.created_at).total_seconds() < 240
```

Comment:
What is this going to be used for?

Reply:
Review-Helper's backend uses it as one of the parameters to decide if it will re-process the patch.